File: app/src/main/java/com/test/shoppingapp/shoppinglist/adapter/ShoppingListPagerAdapter.java
package com.test.shoppingapp.shoppinglist.adapter;


import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.test.shoppingapp.shoppinglist.ItemExtras;
import com.test.shoppingapp.shoppinglist.fragment.ItemFragment;

public class ShoppingListPagerAdapter extends FragmentPagerAdapter {

    private String[] tabTitles = new String[]{"All", "Men", "Women"};


    public ShoppingListPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        return ItemFragment.newInstance(new ItemExtras(position));
    }

    @Override
    public int getCount() {
        return tabTitles.length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }
}

File: app/src/main/java/com/test/shoppingapp/shoppinglist/adapter/ShoppingListPagerAdapter.kt
package com.test.shoppingapp.shoppinglist.adapter


import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.test.shoppingapp.shoppinglist.ItemExtras
import com.test.shoppingapp.shoppinglist.fragment.ShoppingListFragment

class ShoppingListPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val tabTitles = arrayOf("All", "Men", "Women")

    override fun getItem(position: Int): androidx.fragment.app.Fragment {
        return ShoppingListFragment.newInstance(ItemExtras(position))
    }

    override fun getCount(): Int {
        return tabTitles.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return tabTitles[position]
    }
}
